File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function (root) {
    const dfs = function (node, high) {
        if (node === null) {
            return 0;
        }

        let count = 0;
        if (node.val >= high) {
            count += 1;
            high = node.val;
        }

        return dfs(node.left, high) + dfs(node.right, high) + count;
    }

    let result = dfs(root, root.val);

    return result;
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node, high):
            if node is None:
                return 0

            count = 0
            if node.val >= high:
                count += 1
                high = node.val

            return dfs(node.left, high) + dfs(node.right, high) + count

        result = dfs(root, root.val)

        return result
